src/sentry/rules/actions/utils.py

```diff
@@ -0,0 +1,116 @@
+from collections import defaultdict
+from typing import Any, DefaultDict
+
+from sentry.api.serializers.models.rule import generate_rule_label
+from sentry.models.environment import Environment
+from sentry.models.rule import Rule
+
+
+def get_updated_rule_data(rule: Rule) -> dict[str, Any]:
+    rule_data = dict(rule.data)
+    if rule.environment_id:
+        rule_data["environment_id"] = rule.environment_id
+    if rule.owner:
+        rule_data["owner"] = rule.owner
+    rule_data["label"] = rule.label
+    return rule_data
+
+
+def check_value_changed(
+    rule_data: dict[str, Any], rule_data_before: dict[str, Any], key: str, word: str
+) -> str:
+    if rule_data.get(key) != rule_data_before.get(key):
+        old_value = rule_data_before.get(key)
+        new_value = rule_data.get(key)
+        return f"Changed {word} from *{old_value}* to *{new_value}*"
+    return ""
+
+
+def check_added_or_removed(
+    rule_data_after: dict[str, Any],
+    rule_data_before: dict[str, Any],
```

Comment:
nit: I think these can be something more generic, like `prior_state` and `present_state`